packages/scripts/src/create-node-entrypoint.ts

```
@@ -73,13 +73,27 @@ import { workerData } from "node:worker_threads";
 const options = workerData?.runtimeOptions ?? parseRuntimeOptions();
 const verbose = options.get('verbose') ?? false;
 const envId = options.get('environment_id') ?? 'unknown_environment';
+
 if (verbose) {
     console.log('[${env.name}]: Started with options: ', options);
 }
 
 const unbindMetricsListener = bindMetricsListener();
-
-main({
+const unbindTerminationListener = bindRpcListener('terminate', async () => {
+    if (verbose) {
+        console.log('[${env.name}]: Termination Requested. Waiting for engine.');
+    }
+    unbindTerminationListener();
+    unbindMetricsListener();
+    try {
+        const engine = await running;
+        console.log('[${env.name}]: Terminating');
+        return engine.shutdown();
+    } catch (e) {
+        return;
```

would this not swallow shutdown errors we would like to see?